### PR TITLE
Use one http client per filter,  use custom executor service

### PR DESCRIPTION
**Pull request by ternovoy**
@alexeytokar @rasmus93 @maxim-vorotilo

### CODE REVIEW

File: filters/src/main/java/ua/net/tokar/json/rainbowrest/BasicHttpClient.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BasicHttpClient implements HttpClient {
+    private org.apache.http.client.HttpClient httpClient;
```
Comment (rasmus93) [inline filters/src/main/java/ua/net/tokar/json/rainbowrest/BasicHttpClient.java]: seems like you don't need package name here

File: filters/src/main/java/ua/net/tokar/json/rainbowrest/BasicHttpClient.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BasicHttpClient implements HttpClient {
```
Comment (alexeytokar) [inline filters/src/main/java/ua/net/tokar/json/rainbowrest/BasicHttpClient.java]: ApacheHttpClientAdapter

File: filters/src/main/java/ua/net/tokar/json/rainbowrest/BasicHttpHeader.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package ua.net.tokar.json.rainbowrest;
+
+public class BasicHttpHeader implements HttpHeader {
```
Comment (alexeytokar) [inline filters/src/main/java/ua/net/tokar/json/rainbowrest/BasicHttpHeader.java]: why do you need the interface?

**Review by alexeytokar — CHANGES_REQUESTED** [review]

File: filters/src/main/java/ua/net/tokar/json/rainbowrest/BasicHttpClient.java (hunk 1 above)
Comment (ternovoy) [reply filters/src/main/java/ua/net/tokar/json/rainbowrest/BasicHttpClient.java]: We need it, because it's our HttpClient

**Review by alexeytokar — APPROVED** [review]

**Pull request merged**